pkg/config/spec.go

```diff
@@ -277,13 +277,18 @@ func NewUpgradeSpec(cfg *Config) (*v1.UpgradeSpec, error) {
 			recMnt = constants.TransitionDir
 		}
 
+		recoverySrc := cfg.Install.Recovery.Source
```

Comment:
I wonder if `cfg.Install` or `cfg.Install.Recovery` could be `nil` at this point. That would panic.

Reply:
Added  a check